File: src/main/java/org/chousik/collection/builder/collectors/CountryCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.Country;
import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class CountryCollector extends EnumCollector<Country> {
    @Override
    public Country ask(String name, IValidator validator) throws ScriptExecutionError {
        return askEnum(name, validator, Country::valueOf, Country::getValue);
    }
}

File: src/main/kotlin/collection/builder/collectors/CountryCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.Country
import org.chousik.collection.validators.IValidator
class CountryCollector : EnumCollector<Country>() {

    override fun ask(name: String, validator: IValidator<String?>): Country {
        return askEnum(
            name, validator,
            { value: String -> Country.valueOf(value) }, Country.value
        )
    }
}